### PR TITLE
Remove pre-generated SYSCONF

### PR DESCRIPTION
**Pull request by leoetlino**
Dolphin is able to generate one with all correct default settings, so we don't need to ship with a pre-generated SYSCONF and worry about syncing default settings.

Additionally, this commit changes SysConf to work with session SYSCONFs so that Dolphin is able to generate a default one even for Movie/TAS. Which SYSCONF needs to be touched is explicitly specified to avoid confusion about which file SysConf is managing.

---
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/4621)

### CODE REVIEW

File: Source/Core/Common/SysConf.h [hunk 1]
```diff
@@ -74,7 +74,12 @@ struct SSysConfEntry
 class SysConf
 {
 public:
-  SysConf();
+  enum class WiiRootType
```
Comment (JosJuice) [inline Source/Core/Common/SysConf.h]: We already have an enum like this: https://github.com/dolphin-emu/dolphin/blob/3cfc49a613a0f8e137c251694c97f0f3589f3216/Source/Core/Common/NandPaths.h#L21

File: Source/Core/Common/SysConf.cpp [hunk 2]
```diff
@@ -430,11 +430,10 @@ void SysConf::UpdateLocation()
   // Clear the old filename and set the default filename to the new user path
   // So that it can be generated if the file does not exist in the new location
   m_Filename.clear();
-  // Note: We don't use the dummy Wii root here (if in use) because this is
-  // all tied up with the configuration code.  In the future this should
-  // probably just be synced with the other settings.
+  // In the future the SYSCONF should probably just be synced with the other settings.
   m_FilenameDefault =
-      File::GetUserPath(D_WIIROOT_IDX) + DIR_SEP WII_SYSCONF_DIR DIR_SEP WII_SYSCONF;
+      File::GetUserPath(root_type == WiiRootType::User ? D_WIIROOT_IDX : D_SESSION_WIIROOT_IDX) +
```
Comment (JosJuice) [inline Source/Core/Common/SysConf.cpp]: You should probably make it possible to call the following function from here: https://github.com/dolphin-emu/dolphin/blob/3cfc49a613a0f8e137c251694c97f0f3589f3216/Source/Core/Common/NandPaths.cpp#L62

**Comment by JMC47** [comment]
It no longer hangs on shutdown, but it is instant. I can't tell if graceful shutdown is working at a casual glance.

**Comment by leoetlino** [comment]
Dolphin always attempts to trigger a shutdown by STM with most Wii titles, so if it no longer hangs, then it's working.

**Comment by JMC47** [comment]
STM shutdown is not working properly on netplay.  Verified by turning on confirm on stop.  Both the initial confirm and the warning for loss of data come up when I hit the X.

**Comment by leoetlino** [comment]
I have a fix for this, but it doesn't really fit into this PR imo…

File: Source/Core/Common/SysConf.cpp [hunk 3]
```diff
@@ -76,20 +75,25 @@ bool SysConf::LoadFromFile(const std::string& filename)
...
     }
   }
 
   return false;
 }
 
+// Apply Wii settings from normal SYSCONF on Movie recording/playback
+void SysConf::ApplySettingsFromMovie()
+{
+  if (!Movie::IsRecordingInput() && !Movie::IsPlayingInput())
```
Comment (JosJuice) [inline Source/Core/Common/SysConf.cpp]: This can be replaced with `if (!Movie::IsMovieActive())`

Comment (leoetlino) [inline Source/Core/Common/SysConf.cpp]: Done.

**Pull request merged**